### PR TITLE
Add punctuators

### PR DESCRIPTION
**Pull request by katcipis**
Adding all punctuators found here: http://es5.github.io/#x7.6.1.1

### CODE REVIEW

File: token/token.go [hunk 1]
```diff
@@ -18,12 +18,52 @@ const (
...
+	TEqual // stands for triple equal x_x
+	NotTEqual
+	LShift
+	RShift
+	RShiftZero // Right Shift with zero filling, yay JS
+	And
+	Or
+	Xor
+	Not
+	LNot
```
Comment (tiago4orion) [inline token/token.go]: LNot ?

Comment (ryukinix) [reply token/token.go]: Logical Not maybe (!).
`Not` probably means the bitwise Not (~).

File: lexer/lex.go [hunk 2]
```diff
@@ -88,57 +92,77 @@ func (l *lexer) initialState() (Tokval, lexerState) {
...
+		rune(']'):  state(token.RBrack),
+		rune('{'):  state(token.LBrace),
+		rune('}'):  state(token.RBrace),
+		rune('<'):  state(token.Less),
+		rune('>'):  state(token.Greater),
+		rune('&'):  state(token.And),
+		rune('|'):  state(token.Or),
+		rune('^'):  state(token.Xor),
+		rune('~'):  state(token.Not),
+		rune('!'):  state(token.LNot),
```
Comment (ryukinix) [inline lexer/lex.go]: Yes, LNot here @tiago4orion.

File: token/token.go (hunk 1 above)
Comment (katcipis) [reply token/token.go]: Indeed logical not, I copied the names from Go's scanner as far as I can remember =P

**Pull request merged**